activesupport/lib/active_support/core_ext/string/conversions.rb

```diff
@@ -18,7 +18,8 @@ class String
   #   "12/13/2012".to_time               # => ArgumentError: argument out of range
   def to_time(form = :local)
     parts = Date._parse(self, false)
-    return if parts.empty?
+    used_keys = %i(year mon mday hour min sec sec_fraction offset)
```

Comment:
Can you move the valid keys array to a constant called `VALID_DATE_PARSE_KEYS` - that way we don't allocate the a new array every time we call `to_time` on a string.